pbsync/pbsync.py

```diff
@@ -413,27 +413,46 @@ def main(argv):
         os.chdir(str(args.debugpath))
 
     # Parser function object for PBSync config file
-    def pbsync_config_parser_func(root): return {
-        'supported_git_version': root.find('git/version').text,
-        'supported_lfs_version': root.find('git/lfsversion').text,
-        'supported_gcm_version': root.find('git/gcmversion').text,
-        'supported_gcm_version_suffix': root.find('git/gcmversionsuffix').text,
-        'gcm_download_suffix': root.find('git/gcmsuffix').text,
-        'expected_branch_name': root.find('git/expectedbranch').text if args.debugbranch is None else str(args.debugbranch),
-        'git_url': root.find('git/url').text,
-        'checksum_file': root.find('git/checksumfile').text,
-        'log_file_path': root.find('log/file').text,
-        'ue4v_user_config': root.find('versionator/userconfig').text,
-        'ue4v_ci_config': root.find('versionator/ciconfig').text,
-        'ue4v_default_bundle': root.find('versionator/defaultbundle').text,
-        'ue4v_ci_bundle': root.find('versionator/cibundle').text,
-        'engine_base_version': root.find('project/enginebaseversion').text,
-        'uproject_name': root.find('project/uprojectname').text,
-        'defaultgame_path': root.find('project/defaultgameinipath').text,
-        'dispatch_config': root.find('dispatch/config').text,
-        'dispatch_drm': root.find('dispatch/drm').text,
-        'dispatch_stagedir': root.find('dispatch/stagedir').text
-    }
+    def pbsync_config_parser_func(root):
+        config_args_map = {
+            'supported_git_version': ('git/version', None),
+            'supported_lfs_version': ('git/lfsversion', None),
+            'supported_gcm_version': ('git/gcmversion', None),
+            'supported_gcm_version_suffix': ('git/gcmversionsuffix', None),
+            'gcm_download_suffix': ('git/gcmsuffix', None),
+            'expected_branch_name': ('git/expectedbranch', None if args.debugbranch is None else str(args.debugbranch)),
+            'git_url': ('git/url', None),
+            'checksum_file': ('git/checksumfile', None),
+            'log_file_path': ('log/file', None),
+            'ue4v_user_config': ('versionator/userconfig', None),
+            'ue4v_ci_config': ('versionator/ciconfig', None),
+            'ue4v_default_bundle': ('versionator/defaultbundle', None),
+            'ue4v_ci_bundle': ('versionator/cibundle', None),
+            'engine_base_version': ('project/enginebaseversion', None),
+            'uproject_name': ('project/uprojectname', None),
+            'defaultgame_path': ('project/defaultgameinipath', None),
+            'dispatch_config': ('dispatch/config', None),
+            'dispatch_drm': ('dispatch/drm', None),
+            'dispatch_stagedir': ('dispatch/stagedir', None)
+        }
+
+        missing_keys = []
+        config_map = {}
+        for key, val in config_args_map.items():
+            key1, override = val
```

Comment:
I would see more value in this being a fallback/default, rather than an override. If it uses a fallback, then the missing key error can just be logged rather than raising an exception.

Reply:
ack
the reason this ended up being like this, is because it was a bit closer to the original behavior;

but i agree that it can be inverted a bit and have and make it have a default/fallback behavior vs override;

Reply:
thought about it a bit more;
so, just to make sure we're on the same page;

the old behavior for `expected_branch_name` was: 
```
'expected_branch_name': root.find('git/expectedbranch').text if args.debugbranch is None else str(args.debugbranch),
```

i.e. (in pseudo-code)
```
if args.debugbranch is None:
    'expected_branch_name' = root.find('git/expectedbranch').text
else
    'expected_branch_name' = str(args.debugbranch)
```

if you reverse the condition, it looks more like an override, than a default; 
because `args.debugbranch` is preferred if provided;
so [reversing things a bit]:
```
if args.debugbranch [ is not  None]:
    'expected_branch_name' = str(args.debugbranch)
else
    'expected_branch_name' = root.find('git/expectedbranch').text
```

but i am fine to change the default logic;
but i wanted to highlight this subtlety;

Reply:
Ah, I didn't notice your usage for the debug branch arg. Yeah the current implementation seems fine then.